### PR TITLE
add (<&>), fixes issue #53

### PR DESCRIPTION
**Pull request by Profpatsch**

### CODE REVIEW

**Comment by sdiehl** [comment]
```
src/Protolude.hs:576:10:
    The fixity signature for `<$>' lacks an accompanying binding
      (The fixity signature must be given where `<$>' is declared)
```

**Comment by Profpatsch** [comment]
Heh, when “it’s such a small change that I couldn’t make any mistake, so will not setup & compile” it never works.

**Comment by sdiehl** [comment]
Hehe, it happens. 😉

File: src/Protolude.hs [hunk 1]
```diff
@@ -570,6 +570,13 @@ throwTo tid e = liftIO (Control.Exception.throwTo tid e)
 foreach :: Functor f => f a -> (a -> b) -> f b
 foreach = flip fmap
 
+-- | Infix version of foreach.
+--
+-- @<&>@ is to '<$>' what '&' is to '$'.
+infixl 4 <&>
+(<&>) :: Functor f => f a -> (a -> b) -> f b
+(<&>) = foreach
+
```
Comment (PierreR) [inline src/Protolude.hs:579]: I don't know if it matters but for the record, the same operator is implemented in `lens` as such:
```
-- | Infix flipped 'fmap'.
--
-- @
-- ('<&>') = 'flip' 'fmap'
-- @
(<&>) :: Functor f => f a -> (a -> b) -> f b
as <&> f = f <$> as
{-# INLINE (<&>) #-}
```
Is your definition the same performance wise ?

Comment (Profpatsch) [reply src/Protolude.hs:579]: I’d assume it’s the same as `foreach`. Can’t imagine GHC doesn’t do this simple replacement.

**Pull request merged**